### ISSUE TITLE
Browserify and Webpack support

### ISSUE THREAD

**Issue by analog-nico** [request/request-promise#93]
Browserify and Webpack support may be added in the next version. For that we need to work on the points below.

BTW, the next version will drop `.bindCLS(...)` and replace `bluebird` by `any-promise`.

For Browserify:
- [ ] Tests that fully cover Request-Promise's code (`test/spec/browserify.js`, `test/fixtures/browserify/`)
- [ ] Analysis if the code that [freshly requires `request`](https://github.com/request/request-promise/blob/master/lib/rp.js#L14-L33) still makes sense in this environment
- [ ] Sample code that shows a good configuration for Browserify

For Webpack:
- [ ] Tests that fully cover Request-Promise's code (`test/spec/webpack.js`, `test/fixtures/webpack/`)
- [ ] Analysis if the code that [freshly requires `request`](https://github.com/request/request-promise/blob/master/lib/rp.js#L14-L33) still makes sense in this environment
- [ ] Sample code that shows a good configuration for Webpack

**Comment by ryderlee** [comment]
Hi. Wanna ask such issue is saying the package does not support webpack and browserify? I just switch from browserify to webpack and now I am having trouble using request-promise to do http-post. (it works when I was using browserify) So I would like to clarify that.

**Comment by analog-nico** [comment]
Hi @ryderlee , `browserify` and `webpack` are not officially supported. However, many still use `request-promise` with both successfully. Since `request-promise` uses `request` and `request` uses node core libraries the challenge with `webpack` is to include browser versions of the node core libraries in the build. For further info read [this issue](https://github.com/request/request/issues/1529).

**Comment by ngocketit** [comment]
I think it'd be a big shortcoming if `Webpack` and/or `browserify` are not supported as they are essential tools nowadays.

**Comment by analog-nico** [comment]
@ngocketit Absolutely! I'd welcome your contributions in regards to the TODOs at the top. In particular, do you have a working example of a good configuration for Webpack?

**Comment by ngocketit** [comment]
@analog-nico Unfortunately I don't have but if I get it work with `webpack`, I'll link it here.
